### PR TITLE
HPCC-19387 DESDL UI to conform to ESP 2.0

### PR DESCRIPTION
**Pull request by miguelvazq**
Make DESDL U/I compatible with _ver1.4 and greater

Signed-off-by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

File: esp/src/eclwatch/DynamicESDLDefinitionQueryWidget.js [hunk 1]
```diff
@@ -39,10 +53,80 @@ define([
...
+            },
+
+        _onBind: function () {
+            var context = this;
+            var selection = this.grid.getSelected();
+            this.dialog = new Dialog({
+                title: this.i18n.PleaseSelectAServiceToBind,
+                style: "width: 500px;"
+            });
+            this.dialog.show();
```
Comment (GordonSmith) [inline esp/src/eclwatch/DynamicESDLDefinitionQueryWidget.js]: Is this show needed here (also called at the bottom)?

File: esp/src/eclwatch/DynamicESDLDefinitionQueryWidget.js [hunk 2]
```diff
@@ -81,11 +170,21 @@ define([
             this.refreshGrid();
         },
 
+        _saveBinding: function (selection) {
+            WsESDLConfig.PublishESDLBinding({
+                EspProcName: "myesp", //"myesp"
```
Comment (GordonSmith) [inline esp/src/eclwatch/DynamicESDLDefinitionQueryWidget.js]: Remove commented code

File: esp/src/eclwatch/DynamicESDLDefinitionQueryWidget.js [hunk 3]
```diff
@@ -81,11 +170,21 @@ define([
...
+                EspBindingName: "EspBinding",
+                EsdlDefinitionID: this.bindingDropDown.get("value"),
+                Overwrite: this.confirmBindCheckbox.get("value"),
+                EsdlServiceName: "MathService",
+                Config: "<Methods><Method name=" + +"/></Methods>"
```
Comment (GordonSmith) [inline esp/src/eclwatch/DynamicESDLDefinitionQueryWidget.js]: Is there meant to be a "name" inside the `+ +`?

File: esp/src/eclwatch/DynamicESDLQueryWidget.js [hunk 4]
```diff
@@ -90,89 +124,279 @@ define([
...
+                                name = "<a href='#' class='dgrid-row-url'>" + _name + "</a>";
+                            }
+                            return img + "&nbsp;" + name;
+                        },
+                        collapseOnRefresh: false, label: this.i18n.Process, sortable: false
                     })
+                    // CreatedTime: { label: this.i18n.Created, width: 90 },
+                    // PublishBy: { label: this.i18n.PublishedBy, width: 90 },
+                    // LastEditTime: { label: this.i18n.LastEdit, width: 90 },
+                    // LastEditBy: {label: this.i18n.LastEditedBy, width: 90}
```
Comment (GordonSmith) [inline esp/src/eclwatch/DynamicESDLQueryWidget.js]: Remove commented code.

File: esp/src/eclwatch/templates/DynamicESDLDetailsWidget.html [hunk 5]
```diff
@@ -3,21 +3,46 @@
...
-                    <div id="${id}StackController" style="width: 100%" data-dojo-props="containerId:'${id}OverviewTabContainer'" data-dojo-type="dijit.layout.StackController"></div>
-                </div>
-                <div id="${id}OverviewTabContainer" data-dojo-props="region: 'center', tabPosition: 'bottom'" data-dojo-type="dijit.layout.StackContainer">
-                    <div id="${id}Configuration" data-dojo-props="showTitle: true, disabled: false, title: '${i18n.Configuration}'" data-dojo-type="DynamicESDLMethodWidget"></div>
-                    <div id="${id}Definition" data-dojo-props="showTitle: true, disabled: false, title: '${i18n.Definition}'" data-dojo-type="ECLSourceWidget"></div>
+                <div data-dojo-props="region: 'center'" data-dojo-type="dijit.layout.ContentPane">
+                    <h2>
+                        <span id="${id}Id" style="margin-left:20px;" class="bold">${i18n.ID}</span>
+                        <button id="${id}ClippyButton" class="clippy" data-clipboard-target="#${id}Id">
+                            <img src="${dojoConfig.urlInfo.resourcePath}/img/clippy.png" alt="Copy to clipboard">
```
Comment (GordonSmith) [inline esp/src/eclwatch/templates/DynamicESDLDetailsWidget.html]: The alt should probably be in i18n?

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Comment by GordonSmith** [comment]
@miguelvazq also see the lint errors above.

**Review by GordonSmith — APPROVED** [review]
Looks fine.

**Comment by richardkchapman** [comment]
Looks to me like some of the autoformat changes may have been lost?

@GordonSmith

**Comment by GordonSmith** [comment]
@richardkchapman Yes, but its not a big deal, we are all going to be turning on out "Format on save" option and are using matching IDEs going forward (and GitHub has made the ?w=1 option available from a button now,.

**Comment by richardkchapman** [comment]
@GordonSmith Meaning some future PR is also going to look confusing, as well as this one. Is it hard to get this one right?

**Comment by GordonSmith** [comment]
If we can get it right for the beta then yes lets do that.

**Comment by richardkchapman** [comment]
Looks like it's a choice between missing the beta and fixing this issue - so I will merge as-is.

**Pull request merged**